backend/benefit/applications/models.py

```diff
@@ -0,0 +1,243 @@
+    """
+    association_has_business_activities = models.BooleanField(null=True)
+
+    """
+    Store the language of the user filling the application.
+    In case a handler enters data from a paper form, this field must be entered by the admin.
+
+    Notes:
+    Default language is "fi".
+    if language is swedish, then the decision text in Ahjo must be also in swedish
+    if language is english, then an english translation of the decision is included in Ahjo as attachment
+    """
+    applicant_language = models.CharField(
+        choices=APPLICATION_LANGUAGE_CHOICES,
+        default=APPLICATION_LANGUAGE_CHOICES[0][0],
+        max_length=2,
+    )
+
+    co_operation_negotiations = models.BooleanField(null=True)
+    co_operation_negotiations_description = models.CharField(
+        max_length=256,
+        verbose_name=_(
+            "additional information about the ongoing co-operation negotiations"
+        ),
+    )
+
+    apprenticeship_program = models.BooleanField(null=True)
+
+    """
+    After applications are moved to archive, they are hidden from the default view.
+    """
+    archived = models.BooleanField()
+
+    """
+    The type of benefit the applicant is applying for
+    """
+    benefit_type = models.CharField(choices=BenefitType.choices, max_length=64)
+
+    start_date = models.DateField(
+        verbose_name=_("benefit start from date"), null=True, blank=True
+    )
+    end_date = models.DateField(
+        verbose_name=_("benefit end date"), null=True, blank=True
+    )
+
+    def get_available_benefit_types(self):
+        if (
+            self.is_association_application()
+            and not self.association_has_business_activities
+        ):
+            return [BenefitType.SALARY_BENEFIT]
+        else:
+            return [
+                BenefitType.SALARY_BENEFIT,
+                BenefitType.COMMISSION_BENEFIT,
+                BenefitType.EMPLOYMENT_BENEFIT,
+            ]
+
+    """
+    If the value of de_minimis_aid is None, that indicates the applicant has not made the selection.
+    de_minimis_aid is only applicable to applicants with business activities, otherwise it should be None.
+    If value is set to True, then this application must have at least one DeMinimisAid, and if False,
+    then the application must have no DeMinimisAid objects.
+    """
+    de_minimis_aid = models.BooleanField(null=True)
+
+    """
+    In future - add Ahjo batch
+    ahjo_batch = models.ForeignKey("AhjoBatch",
+                              verbose_name=_("ahjo batch"),
+                              related_name="applications",
+                              null=True,
+                              on_delete=models.SET_NULL,
+    )
+    """
+
+    bases = models.ManyToManyField("ApplicationBasis", related_name="applications")
+
+    history = HistoricalRecords(table_name="applications_application_history")
```

Comment:
Is there a reason to set this up manually instead of using the `shared.shared.models.abstract_models.HistoricalModel`?

Reply:
I wanted to set the table_name for it, since I need to add the prefix (which, I notice now,  is missing ... fixed)

Reply:
Ah yep, makes sense now.